esp-hal/src/analog/adc/riscv.rs

```diff
@@ -706,10 +715,30 @@ pub(crate) mod asynch {
         task::{Context, Poll},
     };
 
+    #[cfg(esp32c3)]
+    use portable_atomic::{AtomicU32, Ordering};
     use procmacros::handler;
 
     use crate::{asynch::AtomicWaker, peripherals::APB_SARADC, Async};
 
+    #[cfg(esp32c3)]
+    static ASYNC_ADC_COUNT: AtomicU32 = AtomicU32::new(0);
```

Comment:
maybe I'm missing something here but why only for ESP32-C3?

Reply:
I think that's the only chip with 2 ADCs, for the rest we don't need to count to 1

Reply:
the only RISCV - true

Mind adding a comment since it's (at least to me) not obvious

Reply:
that's for sure much better 👍